changes/2020-05-13_remove-keyring-trace.md

```diff
@@ -0,0 +1,186 @@
+# Remove Keyring Trace
+
+## Affected Features
+
+This serves as a reference of all features that this change affects.
+
+| Feature                                                                                                                                                            |
+| ------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| [Keyring Trace](https://github.com/awslabs/aws-encryption-sdk-specification/blob/61f9edd7c4adf8e8ff9af77bbe9eaf3015099a88/framework/structures.md#keyring-trace-2) |
+| [Keyring Failure Communication](https://github.com/awslabs/aws-encryption-sdk-specification/issues/40)                                                             |
+
+## Affected Specifications
+
+This serves as a reference of all specification documents that this change affects.
+
+| Specification                                                              |
+| -------------------------------------------------------------------------- |
+| [Structures](../framework/structures.md)                                   |
+| [Keyring Interface](../framework/keyring-interface.md)                     |
+| [AWS KMS Keyring](../framework/kms-keyring.md)                             |
+| [Raw AES Keyring](../framework/raw-aes-keyring.md)                         |
+| [Raw RSA Keyring](../framework/raw-rsa-keyring.md)                         |
+| [Cryptographic Materials Manager Interface](../framework/cmm-interface.md) |
+| [Default Cryptographic Materials Manager](../framework/default-cmm.md)     |
+
+## Affected Implementations
+
+This serves as a reference for all implementations that this change affects.
+
+| Language   | Version Introduced | Version Removed | Repository                                                                            |
+| ---------- | ------------------ | --------------- | ------------------------------------------------------------------------------------- |
+| C          | 0.1.0              | n/a             | [aws-encryption-sdk-c](https://github.com/aws/aws-encryption-sdk-c)                   |
+| Javascript | 0.1.0              | n/a             | [aws-encryption-sdk-javascript](https://github.com/aws/aws-encryption-sdk-javascript) |
```

Comment:
What I had in mind for this is:

1. Change is approved and merged.
1. Implementations are updated.
1. Change doc is updated with "Version Removed" for each implementation.

I would not expect this section to be present for all spec change docs, but I think that we need this for this one because it is removing a feature, and that removal will not necessarily be obvious if you, say, just look at the feature doc.

Reply:
Now that this is spelled out with version bumps, I think that's better. There's no implied race between the spec and the implementations, we just need a good way to track the gap between the current version of the spec and the latest implemented version for each language.